services/packagist/packagist-php-version.service.js

```diff
@@ -1,77 +1,69 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const log = require('../../core/server/log')
+const { allVersionsSchema, BasePackagistService } = require('./packagist-base')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class PackagistPhpVersion extends LegacyService {
-  static get category() {
-    return 'platform-support'
-  }
+const { NotFound } = require('..')
 
+module.exports = class PackagistPhpVersion extends BasePackagistService {
   static get route() {
     return {
       base: 'packagist/php-v',
-      pattern: ':user/:repo',
+      pattern: ':user/:repo/:version?',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'php',
+      color: 'blue',
     }
   }
 
+  async handle({ user, repo, version = 'dev-master' }) {
+    const allData = await this.fetch({
+      user,
+      repo,
+      schema: allVersionsSchema,
+    })
+    try {
+      return this.constructor.render({
+        php: allData.package.versions[version].require.php,
+      })
+    } catch (e) {
+      throw new NotFound({ prettyMessage: 'invalid version' })
```

Comment:
What is this `try`/`catch` catching? Could this be done using Joi instead? Let's use Joi for validation whenever we can, and avoid wrapping `render()` in a `try`/`catch`.

Reply:
Agreed

Reply:
Joi will match and return all valid versions, but it will throw a TypeError if you try to use an invalid version. Do you think we should do something like this?

```
if (!allData.package.versions.hasOwnProperty(version)) {
    throw new NotFound({ prettyMessage: 'invalid version' })
}
```

Reply:
pushed.